src/test/java/org/junit/tests/assertion/AssertionTest.java

```diff
@@ -83,11 +98,29 @@ public void arraysActualNullMessage() {
     }
 
     @Test
-    public void arraysDifferentLengthMessage() {
+    public void arraysDifferentLengthDifferingAtStartMessage() {
+        try {
+            assertArrayEquals("not equal", (new Object[]{true}), (new Object[]{false, true}));
+        } catch (AssertionError exception) {
+            assertEquals("not equal: array lengths differed, expected.length=1 actual.length=2; arrays first differed at element [0]; expected:<true> but was:<false>", exception.getMessage());
+        }
```

Comment:
I saw that, too. Although I do like the idea of a test only testing one thing, I don't like tests that can pass for the wrong reason. Also, the way the tests are structured, you have to write two tests for each use case (violating Don't Repeat Yourself). Finally, if the tests were written the way I suggest, they would only be testing one thing: verifying that with the given input, `assertArrayEquals()` throws the expected exception.

`Assert.expectThrows ()` would be perfect here if we could use lamdas. We could use it here without lamdas, or we could add the `return` and `fail()` calls.